budgeteer-admin-interface/src/main/java/de/adesso/budgeteer/web/pages/administration/EditProjectPage.java

```diff
@@ -0,0 +1,231 @@
+                    success(getString("project.saved"));
+                }
+            }
+        };
+        form.add(new TextField<>("projectTitle", model(from(form.getModelObject()).getName())));
+        DateRange defaultDateRange = new DateRange(DateUtil.getBeginOfYear(), DateUtil.getEndOfYear());
+        form.add(new DateRangeInputField("projectStart", model(from(form.getModelObject()).getDateRange()), defaultDateRange, DateRangeInputField.DROP_LOCATION.DOWN));
+        return form;
+    }
+
+    private ListView<User> createUserList(String id, IModel<List<User>> model) {
+        User thisUser = BudgeteerSession.get().getLoggedInUser();
+        long projectID = getParameterId();
+        return new ListView<User>(id, model) {
+
+            @Override
+            protected void populateItem(final ListItem<User> item) {
+                item.add(new Label("username", model(from(item.getModel()).getName())));
+                Link deleteButton = new Link("deleteButton") {
+                    @Override
+                    public void onClick() {
+                        setResponsePage(new DeleteDialog() {
+                            @Override
+                            protected void onYes() {
+                                if(thisUser.getId() == item.getModelObject().getId()){
+                                    userService.removeUserFromProject(projectID, item.getModelObject().getId());
+                                }else{
+                                    userService.removeUserFromProject(projectID, item.getModelObject().getId());
+                                }
+                                setResponsePage(EditProjectPage.class, EditProjectPage.this.getPageParameters());
+                            }
+
+                            @Override
+                            protected void onNo() {
+                                setResponsePage(EditProjectPage.class, EditProjectPage.this.getPageParameters());
+                            }
+
+                            @Override
+                            protected String confirmationText() {
+                                return new StringResourceModel("delete.user.from.project.confirmation", EditProjectPage.this).getString();
+                            }
+                        });
+                    }
+                };
+
+                List<UserRole> choices = Arrays.asList(UserRole.values());
+                ListMultipleChoice<UserRole> makeAdminList = new ListMultipleChoice<>("roleDropdown", new Model<>(
+                        new ArrayList<>(item.getModelObject().getRoles(projectID))), choices);
+                HashMap<String, String> options = new HashMap<>();
+                options.clear();
+                options.put("buttonWidth","'120px'");
+                makeAdminList.add(new MultiselectBehavior(options));
+                makeAdminList.add(new AjaxFormComponentUpdatingBehavior("change") {
```

Comment:
Please refactor and create own components.

Reply:
Done